src/components/interface/perf_test_read/perf_test_read.h

```diff
@@ -0,0 +1,6 @@
+#ifndef PERF_TEST_READ_H
```

Comment:
isn't this the same interface as pong?  If so, I don't think that we need a second, identical interface.